pkg/scheduler/framework/interface.go

```diff
@@ -158,7 +158,14 @@ func (s *Status) Message() string {
 	if s == nil {
 		return ""
 	}
-	return strings.Join(s.reasons, ", ")
+	return strings.Join(s.Reasons(), ", ")
+}
+
+// WithError sets the given `err` to s.error,
+// and returns the given status object.
+func (s *Status) WithError(err error) *Status {
```

Comment:
I think we don't need this function.

Reply:
We still need it; otherwise, we don't have lib to compose a non-error Status with FitError - which is a typical usage for Filter / PostFilter:

```go
# PostFilter
return framework.NewPostFilterResultWithNominatedNode(""), framework.NewStatus(framework.Unschedulable).WithError(fitError)
```

Reply:
Ack